### DISCUSSION TITLE
Jump Drive Confusion

### DISCUSSION THREAD

**Discussion by SultanRTG** [endless-sky/endless-sky#9290] [Q&A] [answered]
When you first get a ship, it's equipped with a hyperdrive. Through the various tidbits of knowledge, you know that there is a reason that your hyperdrive can only go to certain systems in certain ways because of hyperlanes. The Jump Drive can bypass hyperlanes, but still can only travel to certain systems. For example, why can the Jump Drive go to Deneb, but not Tau Ceti?

**Comment by bene-dictator** [comment, marked as answer]
Tau Ceti is simply a star that MZ didn't put on the map, among others like Barnard's Star, Thuban, Kang, Eltanin, Atlas, Prima Hyadum, Epsilon Eridani, Wolf 359, Helvetios, Cervantes...

Additionally, Endless Sky isn't trying to put every star on the map and won't be able to anyway since the map is a 2D plane that's trying to represent 3D space. Not without making everything impossible to read or implement, anyway. See this map of all stars within 50ly (https://projectrho.com/public_html/starmaps/media/winchhabhyg/HabHYG50ly_Large.png). There's just under 1000 stars in that image – and Deneb is around 2,615 ly away, so where do you even start? Not much I can do about that, but you'll have to accept that Deneb is as close as Sirius (~10 ly) in ES.

But really there is no reason at all, we are not looking to chart every star there is and we can't possibly do so. We can try to expand the stars we have, and a couple people have tried that, but it would be difficult to get into the game itself.

**Reply by Zireael07** [reply]
Nitpick: it's definitely possible to chart stars closest (and therefore with names most recognizable to players) to Sol. (I say this as someone making a 2D space game with a 2D star map...)

**Reply by bene-dictator** [reply]
Not accurately is what I mean, and not without some wild-looking hyperlanes if you put it into 3D space. I've done it before.

<img width="1440" alt="Screenshot_2023-03-27_at_8 58 06_PM" src="https://github.com/endless-sky/endless-sky/assets/115441627/7e67803c-5e91-4a8a-a37a-c60208bc201c">

All in correct 2D space, but no Z component.

**Reply by Zireael07** [reply]
Neither Endless Sky nor my own game need to be absolutely accurate.

Some hints as to how to map 3d space in 2d: https://www.projectrho.com/public_html/rocket/spacemaps.php 

After several months of waffling, I ended up indicating Z by icon size AND icon color. (Will post a screen as soon as I find one)

**Reply by SultanRTG** [reply]
What about the map system in Naev? It uses a sort of 3d map system.

**Reply by Saugia** [reply]
Naev is another game based off of Escape Velocity, the same series that ES is inspired by. They have their own ways of doing a map, and we have ours. ES is based around having a 2D space map, and we will not be going to 3D. Additionally, as bene says, our goal is not to add every start there is, or attempt to do so in a "3D" view of the galaxy in some method.

In content creation with ES, vanilla systems should not overlap at the farthest zoom where names are visible. This might be different if we decide to add systems to the core, but otherwise that is the general rule of map systems.

To answer the topic's question, JD is based on the 2D map and a specific circular distance around a given system. There is no 3D nor do we plan on it, and we don't plan on adding systems that are either not present currently or are involved in some sort of 3D view.

**Reply by Zireael07** [reply]
There is another way to have a fully 2D map that has closest star systems, while having reasonable distances. See https://starmap2d.appspot.com and their source code https://github.com/filiph/starmap2d (they used a Kohonen self organizing map, i.e. a neural network, but I'm reasonably certain you can achieve the same thing with a "force directed spring layout" algorithm that many graph libraries use.

**Comment by ifedbro** [comment]
its  important. thanks
